langchain/document_loaders/json_loader.py

```diff
@@ -0,0 +1,86 @@
+"""Loader that loads data from JSON."""
+import json
+from pathlib import Path
+from typing import Callable, Dict, List, Union
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseLoader
+
+
+class JSONLoader(BaseLoader):
+    """Loads a JSON file and references a jq schema provided to load the text into
+    documents.
+
+    Example:
+        [{"text": ...}, {"text": ...}, {"text": ...}] -> schema = .[].text
+        {"key": [{"text": ...}, {"text": ...}, {"text": ...}]} -> schema = .key[].text
+        ["", "", ""] -> schema = .[]
+    """
+
+    def __init__(
+        self,
+        file_path: Union[str, Path],
+        jq_schema: str,
+        content_key: str = None,
+        metadata_func: Callable[Dict, Dict] = None,
+    ):
+        """Initialize the JSONLoader.
+
+        Args:
+            file_path (Union[str, Path]): The path to the JSON file.
+            jq_schema (str): The jq schema to use to extract the data or text from the JSON.
+            content_key (str): The key to use to extract the content from the JSON if the jq_schema results to a list of objects (dict).
+            metadata_func (Callable[Dict, Dict]): A function that takes in the JSON object extracted by the jq_schema and the default metadata and returns a dict of the updated metadata.
+        """
+        try:
+            import jq  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "jq package not found, please install it with " "`pipenv install jq`"
+            )
+
+        self.file_path = Path(file_path).resolve()
+        self._jq_schema = jq.compile(jq_schema)
+        self._content_key = content_key
+        self._metadata_func = metadata_func
+
+    def load(self) -> List[Document]:
+        """Load and return documents from the JSON file."""
+
+        data = self._jq_schema.input(json.loads(self.file_path.read_text()))
+
+        # Perform some validation
+        # This is not a perfect validation, but it should catch most cases
+        # and prevent the user from getting a cryptic error later on.
+        if self._content_key is not None:
+            sample = data.first()
+            assert isinstance(sample, dict)
```

Comment:
nit-picky point, but convention in package so far has been to more explicitly validate and raise errors (usually ValueErrors). so something like 
```python
if not isinstance(sample, dict):
  raise ValueError(f'expected foo bar but received baz: {sample}')
```

Reply:
TIL. Updated!